README.md

```diff
@@ -326,6 +251,49 @@ when you want to proxy this traffic. Otherwise, it is optional. See
 [`http.ProxyFromEnvironment`](https://pkg.go.dev/net/http@go1.17.3#ProxyFromEnvironment)
 for possible values.
 
+## Frequently Asked Questions
+
+### Why would I use the proxy?
+
+The proxy removes the need to manage database client certificates and Authorized
+Networks, while providing an encrypted channel to your Cloud SQL instance using
+TLS 1.3 based on ephemeral certificates. In addition, the proxy allow you to manage
+access to your database using IAM.
+
+### Why can't the proxy connect to my private IP instance?
+
+The proxy must have a network path to a private IP instance to function. If no
+network path exists, the proxy cannot create a network path and so cannot connect.
+
+### Is there a library version of the proxy that I can use?
+
+Yes. Cloud SQL supports three language connectors:
+
+- [Cloud SQL Go Connector][go connector]
+- [Cloud SQL Java Connector](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory)
+- [Cloud SQL Python Connector](https://github.com/GoogleCloudPlatform/cloud-sql-python-connector)
+
+If you're using Go, Java, or Python, the language connectors offer the best
+experience. If you're using another language, then the proxy is the way to go.
+
+### Should I use the proxy for large deployments?
+
+We recommend deploying the proxy next to every application. It's possible that
+large deployments may bump up against the SQL Admin API quota. In those cases,
+we recommend deploying the proxy with a connection pooler like [pgbouncer][] or
+[ProxySQL][]. For details, see [Running the Cloud SQL Proxy as a Service][service-example].
+
+### Can I share the proxy across mulitple applications?
+
+Instead of using a single proxy across multiple applications, we recommend using
+one proxy instance for every application process. The proxy uses the context's IAM
+principal and so have a 1-to-1 mapping between application and IAM principal is
+best. If multiple applications use the same proxy instance, then it becomes unclear
+from an IAM perspective which principal is doing what.\*\*\*\*
```

Comment:
I would consider adding a question "How should I use the proxy" 2nd on the list and say something like:

"The Cloud SQL Auth proxy is a gateway to your Cloud SQL instance. We recommend running it as closely to your application as possible, and making sure to secure connections going to the proxy. Here are some common examples of how to run the proxy in different environments:
  -- links to examples
  "